src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1108,6 +1109,60 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       throw failures.failure(info, elementsShouldSatisfy(actual, unsatisfiedRequirements, info));
   }
 
+  /**
+   * Verifies that all given consumers can be satisfied by elements in the iterable under test with
+   * an element at most satisfying one consumer. No order requirement.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Iterable}.
+   * @param consumers the consumers that are expected to be satisfied by the elements of the given {@code Iterable}.
+   * @throws NullPointerException if the given consumers array or var args comsumers is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if any {@code Consumer} in the consumers cannot be satisfied by elements in the given {@code Iterable}.
+   */
+  @SafeVarargs
+  public final <E> void assertSatisfy(AssertionInfo info, Iterable<? extends E> actual, Consumer<? super E>... consumers) {
+    assertNotNull(info, actual);
+    requireNonNull(consumers, "The Consumer<? super E>... expressing the assertions consumers must not be null");
+    for(Consumer<? super E> consumer: consumers)
+      requireNonNull(consumer, "The element in consumers must not be null");
+
+    List<E>[] satisfiedElementsLists = new ArrayList[consumers.length];
+    for (int i = 0; i < consumers.length; i++) {
+      Consumer<? super E> consumer = consumers[i];
+      satisfiedElementsLists[i] = stream(actual).filter(element -> {
+        try {
+          consumer.accept(element);
+        } catch (AssertionError ex) {
+          return false;
+        }
+        return true;
+      }).collect(toList());
+    }
+    if (!isSatisfied(satisfiedElementsLists, 0))
+      throw failures.failure(info, shouldSatisfy(actual));
+  }
+
+  private static <E> boolean isSatisfied(List<E>[] satisfiedElementsLists, int begin) {
+    // recursively test whether we can find any specific matching strategy that can meet the requirements
+    if (begin == satisfiedElementsLists.length) return true;   // all consumers have been satisfied
+    if (satisfiedElementsLists[begin].isEmpty()) return false;   // no satisfied element for consumers[begin]
+
+    for (E element : satisfiedElementsLists[begin]) {
+      List<Integer> indeces = newArrayList();
+      for (int i = begin + 1; i < satisfiedElementsLists.length; i++) {
+        if (satisfiedElementsLists[i].remove(element)) {
+          indeces.add(i);
+        }
+      }
+
+      if (isSatisfied(satisfiedElementsLists, begin + 1)) return true;
+      for (int i : indeces)
```

Comment:
Would the variable name "removedElementRowIndices" be a better fit (also fixing the typo)?

```suggestion
      List<Integer> removedElementRowIndices = newArrayList();
      for (int i = begin + 1; i < satisfiedElementsLists.length; i++) {
        if (satisfiedElementsLists[i].remove(element)) {
          removedElementRowIndices.add(i);
        }
      }

      // Check whether remaining consumers can be satisfied
      if (isSatisfied(satisfiedElementsLists, begin + 1)) return true;
      
      for (int i : removedElementRowIndices)
```